Sources/CoreCommands/SwiftCommandState.swift

```diff
@@ -741,7 +741,7 @@ package final class SwiftCommandState {
             configuration: options.build.configuration,
             toolchain: toolchain,
             triple: triple,
-            flags: options.build.buildFlags,
+            flags: toolsBuild ? .init() : options.build.buildFlags,
```

Comment:
`buildFlags` is a combination of C/C++/Swift/Linker/XCBuild which means that host compiler doesn't get any of them but are there situations when some of the flags intended for the host and/or tools? 

At the very list I think descriptions of all flags and not just `swiftCompilerFlags` have to be updated to indicate that they are no longer applied to the host.